src/DefaultBuilder/src/ConfigureWebHostBuilder.cs

```diff
@@ -53,6 +54,7 @@ public IWebHostBuilder ConfigureAppConfiguration(Action<WebHostBuilderContext, I
                 // We allow changing the web root since it's based off the content root and typically
                 // read after the host is built.
                 _environment.WebRootPath = Path.Combine(_environment.ContentRootPath, value);
+                _environment.WebRootFileProvider = new PhysicalFileProvider(_environment.WebRootPath);
```

Comment:
This change is fine, but it's a bit unfortunate that setting `WebRootPath` doesn't configure the file provider. A casual user is going to have a hard time figuring this out.

Reply:
> This change is fine, but it's a bit unfortunate that setting WebRootPath doesn't configure the file provider. A casual user is going to have a hard time figuring this out.

It used to do this as introduced in https://github.com/dotnet/aspnetcore/pull/34140. But then that code got deleted. I opted to bring in just this to reduce the delta of this change.

Reply:
Should we create the directory if it's empty like we do here
https://github.com/dotnet/aspnetcore/blob/bc0611939b74130fa4295cc2021e632b215b7298/src/Hosting/Hosting/src/Internal/HostingEnvironmentExtensions.cs#L51-L55

Reply:
The bar is "did that work in .NET 5", if yes then we should support it (and yes I did delete that test)

Reply:
Good recommendation! Fixed.